stored_requests/backends/file_fetcher/fetcher.go

```diff
@@ -5,126 +5,133 @@ import (
-	return storedRequests, storedImpressions, errs
+func readJSONFile(dir string, key string) (json.RawMessage, error) {
+	f := path.Join(dir, key) + ".json"
+	if fileData, err := ioutil.ReadFile(f); err != nil {
+		return nil, err
+	} else {
+		return json.RawMessage(fileData), nil
+	}
 }
 
-func (fetcher *eagerFetcher) FetchCategories(ctx context.Context, primaryAdServer, publisherId, iabCategory string) (string, error) {
-	fileName := primaryAdServer
-
-	if len(publisherId) != 0 {
-		fileName = primaryAdServer + "_" + publisherId
+func fetchObjects(dir string, dataType string, ids []string) (jsons map[string]json.RawMessage, errs []error) {
+	jsons = make(map[string]json.RawMessage)
+	for _, id := range ids {
+		if data, err := readJSONFile(dir, id); err == nil {
+			jsons[id] = data
+		} else {
+			errs = append(errs, stored_requests.NotFoundError{
+				ID:       id,
+				DataType: dataType,
+			})
+		}
 	}
+	return jsons, errs
+}
 
-	if fetcher.Categories == nil {
-		fetcher.Categories = make(map[string]map[string]stored_requests.Category)
+func fetchAllObjects(dir string, glob string, dataType string) (jsons map[string]json.RawMessage, errs []error) {
+	var ids, files []string
+	var err error
+	path := filepath.Join(dir, glob)
+	if files, err = filepath.Glob(path); err != nil {
+		return nil, append(errs, fmt.Errorf("Error scanning for %s in %s: %v", dataType, path, err))
 	}
-	if data, ok := fetcher.Categories[fileName]; ok {
-		return data[iabCategory].Id, nil
+	for _, f := range files {
+		fn := filepath.Base(f)
+		id := strings.TrimSuffix(fn, filepath.Ext(fn))
+		ids = append(ids, id)
 	}
+	return fetchObjects(dir, dataType, ids)
+}
 
-	if primaryAdServerDir, found := fetcher.FileSystem.Directories[primaryAdServer]; found {
-
-		if file, ok := primaryAdServerDir.Files[fileName]; ok {
-
-			tmp := make(map[string]stored_requests.Category)
+// FetchRequests fetches the stored requests for the given IDs.
+func (fetcher *fileFetcher) FetchRequests(ctx context.Context, requestIDs []string, impIDs []string) (requests map[string]json.RawMessage, imps map[string]json.RawMessage, errs []error) {
+	requests, reqErrs := fetchObjects(fetcher.StoredRequestsDir, "Request", requestIDs)
+	imps, impErrs := fetchObjects(fetcher.StoredImpsDir, "Imp", impIDs)
+	return requests, imps, append(reqErrs, impErrs...)
+}
 
-			if err := json.Unmarshal(file, &tmp); err != nil {
-				return "", fmt.Errorf("Unable to unmarshal categories for adserver: '%s', publisherId: '%s'", primaryAdServer, publisherId)
-			}
-			fetcher.Categories[fileName] = tmp
-			resultCategory := tmp[iabCategory].Id
-			primaryAdServerDir.Files[fileName] = nil
+// FetchAllRequests returns comprehensive maps containing all the requests and imps in filesystem
+func (fetcher *fileFetcher) FetchAllRequests(ctx context.Context) (requests map[string]json.RawMessage, imps map[string]json.RawMessage, errs []error) {
+	requests, errs = fetchAllObjects(fetcher.StoredRequestsDir, "*.json", "Request")
```

Comment:
Good point. I was planning to take a look at this after #1446 which introduces the DataTypes. The bare strings here come from the previous code, but they are in a few other places not touched by this PR (cache, db fetcher etc) and wanted to avoid starting a change from two places.